.github/workflows/release.yml

```diff
@@ -12,17 +15,19 @@ on:
         type: boolean
         default: true
       test_pypi:
-        description: 'Test PyPi.'
+        description: 'Test PyPI.'
         type: boolean
         default: false
 
 jobs:
   release:
     runs-on: ubuntu-latest
+    permissions:
+      id-token: write
     steps:
       - uses: actions/checkout@v4
         with:
-          ref: ${{ inputs.candidate && 'main' || 'stable' }}
+          ref: ${{ inputs.candidate && 'stable' || 'main' }}
```

Comment:
@amontanez24 Shouldn't this be switched? It should checkout `stable` for the real PyPI, and `main` for the test PyPI

Reply:
No, if we're doing a release candidate, then we are bumping the version and directly merging that to main. If the release is not a candidate, then we are doing the real release which should be from stable.